ecosystem/platform/server/app/services/wallet_creator.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+# Copyright (c) Aptos
+# SPDX-License-Identifier: Apache-2.0
+
+class WalletCreator
+  def create_wallet(wallet:, challenge:, signed_challenge:)
+    public_key_bytes = RbNaCl::Util.hex2bin(wallet.public_key[2..])
+    verify_key = RbNaCl::VerifyKey.new(public_key_bytes)
+
+    begin
+      verify_key.verify(signed_challenge, challenge)
+      wallet.save
+    rescue RbNaCl::BadSignatureError
+      # The signature was invalid; therefore, don't save the wallet.
+    end
+
```

Comment:
I think we'll need to do more parse on the challenge, but this looks okay for now